hathor/simulator/miner.py

```diff
@@ -41,6 +42,8 @@ def __init__(self, manager: 'HathorManager', *, hashpower: float):
         self.clock = manager.reactor
         self.block = None
         self.delayedcall = None
+        self.log = logger.new()
+        self.rng = default_rng()
```

Comment:
I added it as a requirement on the constructor, as to make it harder to forget passing it (using `create_miner` the simulator will inject the rng arg, but one could do `MinerSimulator(manager, manager.rng)` instead) because it has to be explicit. What do you think?

Reply:
Seems good. Thanks!